boards/DIM/Src/App/App_RegenPaddle.c

```diff
@@ -0,0 +1,74 @@
+#include <stdlib.h>
+#include <assert.h>
+#include "App_RegenPaddle.h"
+
+struct RegenPaddle
+{
+    uint32_t (*get_paddle_position)(void);
+    uint32_t lower_deadzone;
+    uint32_t upper_deadzone;
+    uint32_t paddle_position;
+    uint32_t regen;
+};
+
+struct RegenPaddle *App_RegenPaddle_Create(
+    uint32_t (*const get_pedal_position)(void),
+    uint32_t lower_deadzone,
+    uint32_t upper_deadzone)
+{
+    struct RegenPaddle *regen_paddle = malloc(sizeof(struct RegenPaddle));
+
+    assert(regen_paddle != NULL);
+
+    regen_paddle->get_paddle_position = get_pedal_position;
+    regen_paddle->lower_deadzone      = lower_deadzone;
+    regen_paddle->upper_deadzone      = upper_deadzone;
+
+    return regen_paddle;
+}
+
+void App_RegenPaddle_Destroy(struct RegenPaddle *const regen_paddle)
+{
+    free(regen_paddle);
+}
+
+ErrorCode App_RegenPaddle_Tick(struct RegenPaddle *const regen_paddle)
+{
+    const uint32_t paddle_position = regen_paddle->get_paddle_position();
+
+    if (paddle_position > 100)
+    {
+        return ERROR_CODE_OUT_OF_RANGE;
+    }
```

Comment:
Couldn't use invalid args error code because paddle position isn't an argument. Came up with a more generic out of range error code. I feel like this error code could be rephrased as something more concise so suggestions are welcome.